### ISSUE TITLE
Some errors not fit into one line

### ISSUE THREAD

**Issue by qarmin** [charliermarsh/ruff#3520]
ruff 0.0.255

When testing files, I see a lot of oneline error info
```
error: Failed to parse Desktop/33/SetPortInfoFF5.py: invalid syntax. Got unexpected token '__init__' at line 13 column 10
error: Failed to parse Desktop/33/PpmImagePlugin4.py: Got unexpected token 󤄳 at line 133 column 35
error: Failed to parse Desktop/33/DTDValidateError4.py: Got unexpected token 𓸾 at line 4 column 1
error: Failed to parse Desktop/33/_test_fortran3.py: unexpected EOF while parsing at line 38 column 1
error: Failed to parse Desktop/33/netr_DELTA_ID_UNION8.py: Got unexpected token 򍾖 at line 14 column 31
error: Failed to parse Desktop/33/winapi7.py: Got unexpected token 􋉐 at line 33 column 1
error: Failed to parse Desktop/33/rtf3.py: invalid syntax. Got unexpected token Newline at line 90 column 45
error: Failed to parse Desktop/33/NETLOGON_SAM_LOGON_RESPONSE6.py: Got unexpected token 񰅣 at line 57 column 65
error: Failed to parse Desktop/33/decoders7.py: Got unexpected nesting at line 2 column 1
error: Failed to parse Desktop/33/comparisons5.py: Got unexpected token 󯻁 at line 1 column 13
error: Failed to parse Desktop/33/abag1.py: unexpected EOF while parsing at line 17 column 1
error: Failed to parse Desktop/33/ed255194.py: invalid syntax. Got unexpected token '|' at line 52 column 2
error: Failed to parse Desktop/33/Content3.py: unexpected EOF while parsing at line 21 column 0
error: Failed to parse Desktop/33/vesti8.py: Got unexpected token 򻲈 at line 3 column 29
error: Failed to parse Desktop/33/libglib-2.0.so.0.7200.3-gdb0.py: Got unexpected token 󵟷 at line 1 column 1
```
but some files produce multiline errors, which is inconsistent and looks bad:
```
error: Failed to parse fix_except0.py: invalid syntax. Got unexpected token """
    try_stmt< 'try' ':' (simple_stmt | suite)
                  cleanup=(except_clause ':' (simple_stmt | suite))+
                  tail=(['except' ':' (simple_stmt | suite)]
                        ['else' ':' (simple_stmt | suite)]
                        ['finally' ':' (simple_stmt | suite)]) >
    """ at line 39 column 14
fix_except0.py:39:15: E999 SyntaxError: invalid syntax. Got unexpected token """
    try_stmt< 'try' ':' (simple_stmt | suite)
                  cleanup=(except_clause ':' (simple_stmt | suite))+
                  tail=(['except' ':' (simple_stmt | suite)]
                        ['else' ':' (simple_stmt | suite)]
                        ['finally' ':' (simple_stmt | suite)]) >
    """
Found 1 error.

```

Example file - [fix_except0.py.zip](https://github.com/charliermarsh/ruff/files/10972513/fix_except0.py.zip)

**Comment by charliermarsh** [comment]
Not much we can do here as we don't control those errors (they come from LALRPOP, from RustPython). We could omit them entirely, but that seems even worse to me.

**Comment by charliermarsh** [comment]
Maybe omitting them (just showing `SyntaxError`) is correct?

**Comment by MichaReiser** [comment]
We could potentially truncate after the first line. 

One thing that's on my mind is to re-work Diagnostics in general and, as part of that, I would have pushed towards printing diagnostics with more context by default (multiline diagnostics). Ideally, the system would support a "verbose" mode that prints extra context information that may be too verbose by default. 

Why I'm bringing this up. Has it been an intentional decision only to print one-line diagnostics when running the `check` command and if so, what was its motivation?

**Comment by charliermarsh** [comment]
@MichaReiser - It's mostly born out of a "consistent-with-Flake8-by-default" attitude. We do support printing Clippy-like context via the `--show-source` flag (which can also be set in `pyproject.toml`):

![Screen Shot 2023-03-18 at 11 36 43 PM](https://user-images.githubusercontent.com/1309177/226152193-8a74de4c-3ccf-4397-bd10-8a1197daf19e.png)

But, clearly, it's not the default. That's not based on a principled stance other than that Flake8 also supports a `--show-source` flag and also doesn't default to it. We could definitely _consider_ making that the default.

**Comment by jfmengels** [comment]
> We could definitely consider making that the default.

Please do :pray:

**Comment by MichaReiser** [comment]
> > We could definitely consider making that the default.
> 
> 
> 
> Please do :pray:

Hy @jfmengels nice to see you here. I'm reading many of your blog posts to get inspiration for ruff's configuration and behaviour. Thanks for taking to write down your knowledge.

**Comment by jfmengels** [comment]
> I'm reading many of your blog posts to get inspiration for ruff's configuration and behaviour. Thanks for taking to write down your knowledge.

Happy to help! Static analysis is a field that I care about a lot, and I want better tools for everyone. And it's not like there is any competition between linters, since we tend to target different languages 😄 
I'm also trying to learn things from Ruff (performance mostly), though since the implementation of `elm-review` is done in a very different language, I don't know how much I will be able to put to use in practice.

But let's not side-track the initial topic anymore 😅, always happy to discuss static analysis in other places.

**Comment by youknowone** [comment]
Is this came from `ParseErrorType::UnrecognizedToken`? The multiline output is totally unexpected. It must be a bug of RustPython.

**Comment by DimitrisJim** [comment]
> but some files produce multiline errors, which is inconsistent and looks bad

but that specific file just happens to have a multi-line unexpected token (a triple quoted string literal in this case):

```python
    PATTERN  """
    try_stmt< 'try' ':' (simple_stmt | suite)
                  cleanup=(except_clause ':' (simple_stmt | suite))+
                  tail=(['except' ':' (simple_stmt | suite)]
                        ['else' ':' (simple_stmt | suite)]
                        ['finally' ':' (simple_stmt | suite)]) >
    """
```

hence why the output is multi-line.

**Comment by youknowone** [comment]
Ah, that was confusing due to contents of the string. Still it looks like the error message need to be escaped (and prorably shortened too?) when being formatted, is it?

**Comment by DimitrisJim** [comment]
> Ah, that was confusing due to contents of the string

Exactly! That's why I wanted to run locally and check and realized its just part of the source code.

Not sure if we shorten/trim here, we can choose either really. I think it makes more sense to show the token that is invalid in its entirety considering that large-ish tokens like this one are not really the norm.

**Comment by MichaReiser** [comment]
> Maybe omitting them (just showing `SyntaxError`) is correct?

This seems reasonable to me. #3931 adds a new `DisplayParseError` that implements our custom rendering for `SyntaxErrors. We can customize it to change the rendering of `UnrecognizedToken` to only show the message, or render the first line only (truncate with `...`). I can take a stab at it

**Comment by youknowone** [comment]
@DimitrisJim That's reasonable. Probably escaping will be enough.
